apollo-ios/Sources/Apollo/Interceptors/JSONResponseParsingInterceptor.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import Combine
+#if !COCOAPODS
+import ApolloAPI
+#endif
+
+/// An interceptor which parses JSON response data into a `GraphQLResult` and attaches it to the `HTTPResponse`.
+public struct JSONResponseParsingInterceptor: ApolloInterceptor {
+
+  public init() { }
+
+  actor CurrentResult<Operation: GraphQLOperation> {
+    var value: JSONResponseParser<Operation>.ParsedResult? = nil
+
+    func set(_ value: JSONResponseParser<Operation>.ParsedResult) {
+      self.value = value
+    }
+  }
+
+  public func intercept<Request: GraphQLRequest>(
+    request: Request,
+    next: NextInterceptorFunction<Request>
+  ) async throws -> InterceptorResultStream<Request.Operation> {
+
+    let currentResult = CurrentResult<Request.Operation>()
+
+    if let request = request as? UploadRequest<Request.Operation> {
+
```

Comment:
Should this be doing something here, or is it maybe in a future stacked PR?

Reply:
This is an accident. It's removed in a future PR. Thought the change got merged in here, but it's in one of the stacked PRs.